.github/chatmodes/PAW-R1 Understanding Agent.chatmode.md

```diff
@@ -0,0 +1,700 @@
+- Git operations (commit/push/PR creation)
+- Code execution or testing
+- Subjective judgments about code quality
+- Implementation recommendations or suggestions
+
+## Start / Initial Response
+
+Before responding, inspect the invocation context to infer starting inputs:
+
+1. **Check for ReviewContext.md**:
+   - Look in chat context or on disk at `.paw/reviews/PR-<number>/ReviewContext.md` or `.paw/reviews/<branch-slug>/ReviewContext.md`
+   - If present, extract PR Number/Branch, Base Branch, Head Branch, Base Commit, Head Commit, Artifact Paths, and Repository
+   - Treat ReviewContext.md as authoritative for all parameters
+
+2. **Determine Context Type**:
+   - **GitHub Context**: PR URL or number provided → use GitHub MCP tools
+   - **Non-GitHub Context**: No PR reference → verify current branch is checked out, request base branch name
+
+3. **Confirm Parameters**:
+   ```
+   I'll analyze this PR to create comprehensive understanding artifacts.
+
+   Context identified:
+   - PR/Branch: [PR #123 OR branch feature/new-auth]
+   - Base Branch: [main]
+   - Head Branch: [feature/new-auth]
+   - Repository: [owner/repo OR local]
+   - Artifact Path: [.paw/reviews/PR-123/ OR .paw/reviews/feature-new-auth/]
+
+   Proceeding with analysis...
+   ```
+
+4. **If parameters missing**, ask only for what's needed:
+   - GitHub: "Please provide the PR URL or number"
+   - Non-GitHub: "Please confirm the base branch name (the branch these changes will merge into)"
+
+## Process Steps (Detailed Workflow)
+
+### Step 1: Context Gathering & ReviewContext.md Creation
+
+1. **Fetch PR Metadata**:
+   - **GitHub**: Use `mcp_github_pull_request_read` (method: get) for metadata
+     - Extract: number, title, author, state, created date, description, labels, reviewers
+     - Get CI status from `mcp_github_pull_request_read` (method: get_status)
+     - Get changed files from `mcp_github_pull_request_read` (method: get_files)
+   - **Non-GitHub**: Use git commands
+     - Current branch as head, provided branch as base
+     - `git log <base>..<head>` for commits and author
+     - `git diff <base>...<head> --stat` for changed files
+     - CI status: "Not available (non-GitHub context)"
+
+2. **Analyze Changed Files**:
+   - Get file paths, additions, deletions for each changed file
+   - Categorize files using File Categorization heuristics (below)
+   - Calculate total lines changed (excluding mechanical changes)
+   - Identify flags: Large PR (>1000 LOC), mechanical-only, CI failures
+
+3. **Create ReviewContext.md immediately**:
+   - Write to `.paw/reviews/<identifier>/ReviewContext.md`
+   - Use complete template structure (see "ReviewContext.md Template" below)
+   - Include all metadata, file categories table, and flags
+   - This becomes the authoritative parameter source for all review stages
+
+### Step 2: File Categorization & Change Analysis
+
+1. **Categorize Each Changed File**:
```

Comment:
I don't want file categorization. I thought this had been removed; please remove from this chatmode, and any references from the implementation plan and spec.

this goes for mechanical vs semantic. I don't think there's a ton of value in classifying changes in this way, and it will draw attention away from higher value outputs. The focus of the Understanding phase is really the code research and the derived specification. The next stage of the process will be evaluating changes and so no classification is necessary in this stage.